src/main/scala/ophan/google/indexing/observatory/RedirectResolver.scala

```diff
@@ -60,32 +54,39 @@ object RedirectFollower extends RedirectFollower with Logging {
 
   val HTTPRedirectStatusCodes: Set[Int] = Set(301, 302, 303, 307, 308)
 
-  def redirectGivenBy(requestUri: URI, response: HttpResponse[_]): Option[URI] =
-    if (HTTPRedirectStatusCodes.contains(response.statusCode)) getRedirectedURI(requestUri, response.headers) else None
+  def redirectGivenBy(requestUri: URI, response: okhttp3.Response): Option[URI] =
+    if (HTTPRedirectStatusCodes.contains(response.code())) getRedirectedURI(requestUri, response.headers()) else None
 
   // based on https://github.com/openjdk/jdk17/blob/4afbcaf55/src/java.net.http/share/classes/jdk/internal/net/http/RedirectFilter.java#L132
-  private def getRedirectedURI(requestUri: URI, headers: HttpHeaders): Option[URI] =
+  private def getRedirectedURI(requestUri: URI, headers: Headers): Option[URI] =
     headers.location.flatMap { newLocation =>
       Try(URI.create(newLocation)).toOption
     }.map(requestUri.resolve) // redirect could be relative to original URL
 
 
-  extension (headers: HttpHeaders) def location: Option[String] = headers.firstValue("Location").toScala
+  extension (headers: Headers) def location: Option[String] = Option(headers.get("Location"))
 
-  private val httpClient: HttpClient = HttpClient.newBuilder()
-    .followRedirects(Redirect.NEVER)
-    .connectTimeout(ofSeconds(2))
-    .build()
+  private val httpClient: OkHttpClient =
+    new OkHttpClient.Builder()
+      .followRedirects(false)
+      .followSslRedirects(false)
+      .callTimeout(Duration.ofSeconds(5))
+      .build()
 
-  def requestFor(uri: URI): HttpRequest =
-    HttpRequest.newBuilder().uri(uri).timeout(ofSeconds(2))
-      .header("User-Agent", "curl/7.54") // NYT accepts curl, but rejects the default "Java-http-client" User-Agent!
-      .method("HEAD", noBody()).build()
+  def requestFor(uri: URI): Request =
+    new Request.Builder()
+      .url(uri.toURL)
+      .head()
+      .header("User-Agent", "curl/7.54")
```

Comment:
Would it be worth copying the comment across (about NYT rejecting the default java http client user agent) or is it no longer applicable?

Reply:
Good question! I'm kind of in two minds about it, but the reason I decided against the comment there is that there's actually also a test committed that carries the same information:

https://github.com/guardian/google-search-indexing-observatory/blob/7cf7dcb430d7330e8b08fd2a81ba1d3c33052ecf/src/test/scala/ophan/google/indexing/observatory/RedirectResolverTest.scala#L19-L25

A **test** is often better than a **comment**, because a comment can fall out of date, no longer be true (yet _seem_ authoritative, and therefore be actively misleading) - but a test will actually tell us if what we say in the comment is true - does adding 'curl' to the user-agent allow us to resolve a NYT article? And if some decides, hey, why is this `.header("User-Agent", "curl/7.54")` line of code there - let's remove it! - the tests will quickly tell them why that's not a good idea.

Doing a git blame on the line will probably also get them the information, which is also good, because then it specifically places it in context as a-thing-that-was-true-at-a-point-in-time, rather than the eternal this-is-true-right-now vibes that an inline code comment gives.

Reply:
Ah, amazing - I'd missed the test, thanks :)